### PR TITLE
Return InvalidArgError when pydantic validation fails on API objects

### PR DESCRIPTION
**Pull request by Jeadie**
**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- When a user passes in extra parameters (see pydantic `extra: str = "forbid"`) to the body of `/bulk/search/`, they receive an InvalidArgsError.

**What is the current behavior?** (You can also link to an open issue here)
- extra parameters are ignored silently

* **What is the new behavior (if this is a feature change)?**
- Raise InvalidArgsError with body
```
{
        "message": json.dumps(error_messages),
        "code": InvalidArgError.code,
        "type": InvalidArgError.error_type,
        "link": InvalidArgError.link
}
```

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
Yes

* **Related Python client changes** (link commit/PR here)
Not needed.

* **Related documentation changes** (link commit/PR here)
Same

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/tensor_search/api.py [hunk 1]
```diff
@@ -41,12 +44,13 @@ def replace_host_localhosts(OPENSEARCH_IS_INTERNAL: str, OS_URL: str):
                 return replaced_str
     return OS_URL
 
-OPENSEARCH_URL = replace_host_localhosts(
-    os.environ.get("OPENSEARCH_IS_INTERNAL", None),
-    os.environ["OPENSEARCH_URL"])
-
+if __name__ in ["__main__", "api"]:
```
Comment (Jeadie) [inline src/marqo/tensor_search/api.py:48]: This is needed to gate when api.py is imported, but not run as main script (e.g. unit tests).

File: tests/tensor_search/test_bulk_search.py [hunk 2]
```diff
@@ -809,7 +825,7 @@ def test_search_other_types_top_search(self):
             results = tensor_search.bulk_search(
                 marqo_config=self.config, query=BulkSearchQuery(
                     queries=[BulkSearchQueryEntity(
-                        index=self.index_name_1, q=str(to_search), search_method=SearchMethod.LEXICAL,
+                        index=self.index_name_1, q=str(to_search), searchMethod=SearchMethod.LEXICAL,
```
Comment (Jeadie) [inline tests/tensor_search/test_bulk_search.py:828]: uncovered a bug 🤣

File: src/marqo/tensor_search/api.py (hunk 1 above)
Comment (pandu-k) [reply src/marqo/tensor_search/api.py:48]: even during unit tests, it's still not run as the main script right?

File: src/marqo/tensor_search/api.py [hunk 3]
```diff
@@ -80,6 +84,23 @@ def marqo_user_exception_handler(request, exc: MarqoWebError):
         return JSONResponse(content=body, status_code=exc.status_code)
 
 
+@app.exception_handler(RequestValidationError)
+async def validation_exception_handler(request: Request, exc: RequestValidationError) -> JSONResponse:
```
Comment (pandu-k) [inline src/marqo/tensor_search/api.py]: We really need this class to be well tested (resilience for missing dictionary keys and attributes). An error in this class results in an unhelpful 500.

**Comment by pandu-k** [comment]
rerunning tests: https://github.com/marqo-ai/marqo/actions/runs/4558116695

File: src/marqo/tensor_search/api.py (hunk 1 above)
Comment (Jeadie) [reply src/marqo/tensor_search/api.py:48]: correct, but it imports functions from this file now (hence the need to change).

**Comment by Jeadie** [comment]
Will need to update. Most have had a unreloaded marqo running when running unit tests locally

**Comment by Jeadie** [comment]
Unit tests: https://github.com/marqo-ai/marqo/actions/runs/4559621361

**Comment by Jeadie** [comment]
https://github.com/marqo-ai/marqo/actions/runs/4570217900

File: src/marqo/tensor_search/api.py (hunk 1 above)
Comment (pandu-k) [reply src/marqo/tensor_search/api.py:48]: What is the problem that this is solving? Have you encountered this sort of situation? Or can you foresee a problem occurring?

Comment (Jeadie) [reply src/marqo/tensor_search/api.py:48]: Yeah, this is an explicit problem right now importing from `tensor_search/api.py`

**Comment by pandu-k** [comment]
- DIND test: https://github.com/marqo-ai/marqo/actions/runs/4592343621
- Local-OS test: https://github.com/marqo-ai/marqo/actions/runs/4592345433
- s2search-OS test https://github.com/marqo-ai/marqo/actions/runs/4592351838

**Review by pandu-k — APPROVED** [review]

**Review by pandu-k — APPROVED** [review]

**Pull request merged**